tests/functional/i/inconsistent/inconsistent_returns_noreturn.py

```diff
@@ -53,3 +53,30 @@ def bug_pylint_4122_bis(s):
         return n
     except ValueError:
         parser_error_name('parser error')
+
+class ClassUnderTest:
+    def _no_return_method(self) -> typing.NoReturn:
+        sys.exit(1)
+
+    def _does_return_method(self) -> int:
+        return 1
+
+    def bug_pylint_8747(self, s: str) -> int:
+        """Every return is consistent because self._no_return_method hints NoReturn"""
+        try:
+            n = int(s)
+            if n < 1:
+                raise ValueError
+            return n
+        except ValueError:
+            self._no_return_method()
+
```

Comment:
```suggestion

        def bug_pylint_8747_underhanded(self, s: str) -> int:  # [inconsistent-return-statements]
        """Every return is not consistent because self._falsely_no_return_method() returns a value"""
        try:
            n = int(s)
            if n < 1:
                raise ValueError
            return n
        except ValueError:
            self._falsely_no_return_method()
```

Reply:
This suggested test may be revealing a separate false negative?

Pylint _doesn't_ raise `inconsistent-return-statements` in this case.

Reply:
It's surprising that's it's the existing behavior because usually we do not take typing into account and use inference instead (#4813) -- it was the comment I was going to make by reading the title.  I upgraded the suggestion, I guess we can simply add this test case to document the false negative then.

Reply:
Resolved by https://github.com/pylint-dev/pylint/pull/8750/commits/ab7679406c63111bcda01bcdeb92b7a08064e33d